daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/ConcurrentCompiledPackages.scala

```diff
@@ -55,33 +55,49 @@ final class ConcurrentCompiledPackages extends CompiledPackages {
             case Some(pkg_) => pkg_
           }
 
+          // Load dependencies of this package and transitively its dependencies.
+          for (dependency <- pkg.directDeps) {
+            if (!_packages.contains(dependency) && !state.seenDependencies.contains(dependency)) {
```

Comment:
It looks like the error for cyclical dependencies is gone. Is that intentional?

Reply:
Yes, the cyclic check didn't make sense anymore as we could've easily encountered the same package multiple times (A depends on (B, C), B depends on C). We don't really need to worry about this since we cannot construct cycles as package ids are content hashes.